mmv1/third_party/terraform/services/compute/data_source_google_compute_subnetworks_test.go

```diff
@@ -11,6 +11,157 @@ import (
 	"github.com/hashicorp/terraform-provider-google/google/envvar"
 )
 
+func TestAccDataSourceGoogleSubnetworks_update(t *testing.T) {
+	t.Parallel()
+	// Common resource configuration
+	static_prefix := "tf-test"
+	random_suffix := acctest.RandString(t, 10)
+	project := envvar.GetTestProjectFromEnv()
+	region := envvar.GetTestRegionFromEnv()
+
+	// Resource identifier used for content testing
+	id := fmt.Sprintf(
+		"projects/%s/regions/%s/subnetworks",
+		project,
+		region,
+	)
+
+	// Configuration of network resources
+	network := static_prefix + "-network-" + random_suffix
+	subnet_1 := static_prefix + "-subnet-1-" + random_suffix
+	subnet_2 := static_prefix + "-subnet-2-" + random_suffix
+	cidr_1 := "192.168.31.0/24"
+	cidr_2 := "192.168.32.0/24"
+
+	// Configuration map used in test deployment
+	context := map[string]interface{}{
+		"cidr_1":   cidr_1,
+		"cidr_2":   cidr_2,
+		"network":  network,
+		"project":  project,
+		"region":   region,
+		"subnet_1": subnet_1,
+		"subnet_2": subnet_2,
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckGoogleSubnetworksConfig(context),
+				Check: resource.ComposeTestCheckFunc(
+					// Test schema
+					resource.TestCheckResourceAttrSet("data.google_compute_subnetworks.all", "subnetworks.1.purpose"),
+					// Test Content
+					resource.TestCheckResourceAttr("data.google_compute_subnetworks.two", "subnetworks.0.purpose", "PEER_MIGRATION"),
+				),
```

Comment:
in particular, these lines check values for the data source, but the tests should be checking values for the resource. Normally in a resource test that happens with an ImportState step instead of explicit checks.